src/System.IO.Compression/src/System/IO/Compression/DeflateStream.Unix.cs

```diff
@@ -1,13 +1,27 @@
 // Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
+using System.Diagnostics;
+
 namespace System.IO.Compression
 {
     public partial class DeflateStream : Stream
     {
         private static WorkerType GetDeflaterType()
         {
-            return WorkerType.Managed; // TODO: Switch to ZLib once interop is worked out
+            try
+            {
+                // Make any P/Invoke into zlib to ensure we're able to find and use it.
+                // If we are, then use zlib.
+                Interop.zlib.zlibCompileFlags(); 
+                return WorkerType.ZLib;
+            }
+            catch
+            {
+                // Otherwise, fallback to managed implementation if zlib isn't available
+                Debug.Fail("zlib unavailable");
```

Comment:
Yeah.  I was thinking that for now we'd have the Debug.Fail to be warned when we're hitting this case, and then separately we could add an internal flag / InternalsVisibleTo the tests that would allow the tests to force it to fallback (for both Windows and Unix) and then we could add a few tests explicitly for the managed implementation.